spec.md

```diff
@@ -289,6 +290,34 @@ encapsulated within the `data` attribute.
 * Constraints:
   * OPTIONAL
 
+# Minimum Supported Event Size
+
+In order to increase interoperability, all CloudEvent consumers SHOULD accept
+events that follow these rules:
+
+* The number of attributes, including the `data` attribute, does not exceed 100.
+  Each index of a `Map` is considered an attribute.
+* All attribute names are 20 characters or less long.
+* All indexes of `Maps` are 20 characters or less long.
+* `Any` attributes containing a `Binary` or a `String` follow the respective
+  rule below. The `data` attribute is excempt from this rule.
+* All `Binary` attributes do not exceed a size of 2KB.
+* All `String` attributes (including String expressions like `URI-reference` and
+  `Timestamp`) do not exceed a size of 2KB.
+* For a `Map`, the `Any`-typed values follow the rules above.
+* The total size of all attributes (including the `data` attribute) does not
+  exceed 40KB/64KB.
```

Comment:
Why the 40KB/64KB here? Is this padding for base64 encoding binary content?

Reply:
I wrote in the PR description:

> If we want to keep a JSON CloudEvent below 64KB (basically for Azure EventGrid compatibility), the last rule should probably state 40KB. If we're more targeting "around 64KB in efficient wire formats", we can go for 64KB in the last rule.

Anyway, I found the time to check what the worst-case CloudEvent would look like. I think like this: https://gist.github.com/cneijenhuis/2984f933f05b9dc579d101ae2d24cd8f with the `data` filled with a Binary. If I didn't make a blunder, I think 45KB (according to this measurement) should guarantee that no JSON document can reach 64KB, but get pretty close to it. 46KB (according to this measurement) can result in a JSON document above 64KB.

To be precise, the worst-case JSON document with 45KB weighs in at 64527 Bytes, aka 63,01 KB. So there is a bit of wiggle room in case I made an error somewhere 😉 or we have further minor changes to the spec. However, any JSON event consumer can reject JSON at 64KB and be compliant with these rules.